articles/openshift/howto-create-private-cluster.md

```diff
@@ -0,0 +1,148 @@
+
+If you change the DNS nameservers on the cluster VNet, then you will need to issue an update on the cluster with the `properties.RefreshCluster` property set to `true` so that the VMs can be reimaged. This will allow them to pick up the new nameservers.
+
+## Sample Configuration Scripts
+
+### Environment
+
+Fill in the environment variables below as using your own values.
+
+> [!NOTE]
+> The locatin must be set to `eastus2` as this is currently the only supported location.
+
+```
+export CLUSTER_NAME=
+export LOCATION=eastus2
+export TOKEN=$(az account get-access-token --query 'accessToken' -o tsv)
+export SUBID=
+export TENANT_ID=
+export ADMIN_GROUP=
+export CLIENT_ID=
+export SECRET=
+```
+
+### private-cluster.json
+Using the environment variables defined above, here is a sample cluster configuration with private cluster enabled.
+
+```
+{
+   "location": "$LOCATION",
+   "name": "$CLUSTER_NAME",
+   "properties": {
+       "openShiftVersion": "v3.11",
+       "networkProfile": {
+           "vnetCIDR": "10.0.0.0/8",
+           "managementSubnetCIDR" : "10.0.1.0/24"
+       },
+       "authProfile": {
+           "identityProviders": [
+               {
+                   "name": "Azure AD",
+                   "provider": {
+                       "kind": "AADIdentityProvider",
+                       "clientId": "$CLIENT_ID",
+                       "secret": "$SECRET",
+                       "tenantId": "$TENANT_ID",
+                       "customerAdminGroupID": "$ADMIN_GROUP"
+                   }
+               }
+           ]
+       },
+       "masterPoolProfile": {
+           "name": "master",
+           "count": 3,
+           "vmSize": "Standard_D4s_v3",
+           "osType": "Linux",
+           "subnetCIDR": "10.0.0.0/24",
+           "apiProperties": {
+            	"privateApiServer": true
+            }
+       },
+       "agentPoolProfiles": [
+           {
+               "role": "compute",
+               "name": "compute",
+               "count": 1,
+               "vmSize": "Standard_D4s_v3",
+               "osType": "Linux",
+               "subnetCIDR": "10.0.0.0/24"
+           },
+           {
+               "role": "infra",
+               "name": "infra",
+               "count": 3,
+               "vmSize": "Standard_D4s_v3",
+               "osType": "Linux",
+               "subnetCIDR": "10.0.0.0/24"
+           }
+       ],
+       "routerProfiles": [
+           {
+               "name": "default"
+           }
+       ]
+   }
+}
+```
+ ## Deploy a private cluster
+```
+az group create --name $CLUSTER_NAME --location $LOCATION
+ 
+cat private-cluster.json | envsubst | curl -v -X PUT \
+-H 'Content-Type: application/json; charset=utf-8' \
+-H 'Authorization: Bearer '$TOKEN'' -d @- \
+ https://management.azure.com/subscriptions/$SUBID/resourceGroups/$CLUSTER_NAME/providers/Microsoft.ContainerService/openShiftManagedClusters/$CLUSTER_NAME?api-version=2019-10-27-preview
+```
+
+## Diagrams
```

Comment:
I need some guidance on these diagrams.
- Do we want to include them?
- If we do include them, we should add some text. What should we emphasize or call out?

Reply:
I think we should include the private cluster architecture diagram and call out the express route connection and the private link connections to from Azure to the cluster's vnet and the custom dns server.